src/lazy/encoder/binary/v1_0/value_writer.rs

```diff
@@ -621,13 +460,12 @@ mod tests {
 
     fn writer_test(
         expected: &str,
-        mut test: impl FnMut(&mut LazyRawBinaryWriter_1_0<&mut Vec<u8>>) -> IonResult<()>,
+        test: impl FnOnce(&mut LazyRawBinaryWriter_1_0<Vec<u8>>) -> IonResult<()>,
     ) -> IonResult<()> {
         let expected = Element::read_all(expected)?;
-        let mut buffer = Vec::new();
-        let mut writer = LazyRawBinaryWriter_1_0::new(&mut buffer)?;
+        let mut writer = LazyRawBinaryWriter_1_0::new(Vec::new())?;
         test(&mut writer)?;
-        writer.flush()?;
+        let buffer = writer.finish()?;
```

Comment:
🗺️ Top-level writers implement the `SequenceWriter` trait, which has an `end()` method.

Curious to hear others' takes on this: we had a `finish()` method already, but now it also has an `end()`. Can I get rid of `finish()`?

Alternatively: should this be `close()` everywhere instead of `end()` or `finish()`?

Reply:
I think `close()` is a good choice that works well for both.

Reply:
Done! See commit https://github.com/amazon-ion/ion-rust/pull/741/commits/020e078febe7c4384087362f84a1e32b122485f7.